addons/email_confirmation/models.py

```diff
@@ -55,6 +63,7 @@ def create(self, values):
         # user is not active yet, he needs to click a link in the email
         if self.env.context.get('confirm_signup'):
             user.active = False
+            user.active_state = 'not_activated'
```

Comment:
Not necessary since 'not_activated' is the default value here. But maybe it shouldn't be the default?